src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -109,42 +124,53 @@ private ReadFilterLibrary(){ /*no instance*/ }
                     .anyMatch(c -> c.getOperator().consumesReferenceBases() && c.getLength() > 0);
         }}
 
+    /** Filter out reads marked as duplicates. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class NotDuplicateReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! read.isDuplicate();}}
 
+    /** Filter out reads representing secondary alignments. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class NotSecondaryAlignmentReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return !read.isSecondaryAlignment();}}
 
+    /** Filter out reads representing supplementary alignments. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class NotSupplementaryAlignmentReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return !read.isSupplementaryAlignment();}}
 
+    /** Filter out unpaired reads. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class PairedReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return read.isPaired();}}
 
+    /** Filter out reads failing platform/vendor quality checks. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class PassesVendorQualityCheckReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! read.failsVendorQualityCheck();}}
 
+    /** Keep only reads that are properly paired. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class ProperlyPairedReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return read.isProperlyPaired();}}
 
+    /**
+     * Keep reads representing primary alignments.
```

Comment:
I haven't change this one, because this filter has a nomenclature problem. It will be nice to have a `PrimaryAlignmentReadFilter` which actually keeps only reads that are not secondary and not supplementary, but that corresponds to other issue (#2165).

Both removing this filter and/or setting the description to the proposed one will constraint it to do not be implemented in the future in the proper way. I prefer to maintain the warning in that case, to show that something is wrong for now with this filter.